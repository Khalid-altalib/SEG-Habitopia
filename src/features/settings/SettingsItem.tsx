import React from "react";
import { Box, Pressable, Text } from "native-base";

type Props = {
  type: string;
};

function SettingsItem(props: Props) {
  const { type } = props;
  return (
    <Pressable>
      <Text>{type}: placeholder</Text>
    </Pressable>
  );
}

export default SettingsItem;
