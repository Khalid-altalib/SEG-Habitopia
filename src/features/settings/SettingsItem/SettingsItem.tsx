import React from "react";
import {
  Box,
  HStack,
  Input,
  Pressable,
  Text,
  TextArea,
  TextField,
  View,
} from "native-base";
import { useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { TouchableOpacity } from "react-native";
import { RootParams } from "../../../../types";

type Props = {
  type: string;
  value: string;
};
//add placeholder text
//password - enter new, confirme
function SettingsItem(props: Props) {
  const navigation = useNavigation<NativeStackNavigationProp<RootParams>>();
  const { type, value } = props;

  return (
    <Box>
      <HStack display={"flex"} alignItems={"flex-start"}>
        <Box flex={1}>{type} </Box>
        <Box borderBottomWidth={"1px"} paddingBottom="3" flex={3}>
          <TouchableOpacity
            onPress={() =>
              navigation.push("SettingDetails", {
                settingType: type,
                defaultValue: value,
              })
            }
          >
            {type === "Password" ? <Text>*******</Text> : <Text>{value}</Text>}
          </TouchableOpacity>
        </Box>
      </HStack>
    </Box>
  );
}

export default SettingsItem;
