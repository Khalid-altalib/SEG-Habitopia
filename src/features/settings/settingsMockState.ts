import { SettingsState } from "./settingsSlice";

/**
 * The mock state for the settings redux state
 */

const settingsMockState: SettingsState = {
  settings: {
    email: "test@test.com",
    password: "password",
    notifications: true,
    name: "Test User",
    biography: "Hello, I'm Test User.",
  },
  fetchSettings: {
    loading: false,
    error: "",
  },
  setSettings: {
    loading: false,
    error: "",
  },
};

export default settingsMockState;
