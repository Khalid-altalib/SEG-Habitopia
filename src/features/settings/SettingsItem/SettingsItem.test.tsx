import React from "react";
import { render } from "@testing-library/react-native";
import SettingsItem from "./SettingsItem";
import TestingWrapperNavigation from "@app/testingWrapperWithNavigation";
import settingsMockState from "../settingsMockState";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

describe("SettingsItem", () => {
  const entry = { name: "Email" };
  const value = "email@email.com";

  const mockState = {
    settings: settingsMockState,
  };

  const mockStore = configureStore([thunk])(mockState);

  it("renders correctly", () => {
    const wrapper = render(
      <TestingWrapperNavigation store={mockStore}>
        <SettingsItem type="text" entry={entry} value={value} />
      </TestingWrapperNavigation>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
