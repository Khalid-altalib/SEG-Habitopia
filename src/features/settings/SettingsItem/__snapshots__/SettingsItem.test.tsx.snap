// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsItem renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        collapsable={false}
        forwardedRef={[Function]}
        gestureResponseDistance={
          {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        nativeBackButtonDismissalEnabled={false}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onHeaderBackButtonClicked={[Function]}
        onNativeDismissCancelled={[Function]}
        onTransitionProgress={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        stackPresentation="push"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        swipeDirection="horizontal"
      >
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backgroundColor="rgb(255, 255, 255)"
          color="rgb(0, 122, 255)"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={false}
          hideBackButton={false}
          largeTitleHideShadow={false}
          title="TestScreen"
          titleColor="rgb(28, 28, 30)"
          topInsetEnabled={false}
          translucent={false}
        />
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            {
              "flex": 1,
              "flexDirection": "column-reverse",
            }
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                },
                {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
                undefined,
              ]
            }
          >
            <View
              dataSet={{}}
              style={{}}
            >
              <View
                dataSet={{}}
                style={
                  {
                    "alignItems": "flex-start",
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "borderBottomWidth": 1,
                      "borderColor": "#d4d4d8",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#18181b",
                        },
                        {
                          "fontFamily": undefined,
                          "fontSize": 15,
                        },
                        {
                          "flex": 1,
                        },
                      ]
                    }
                  >
                    Email
                  </Text>
                  <View
                    dataSet={{}}
                    style={
                      {
                        "flex": 3,
                        "paddingBottom": 12,
                      }
                    }
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "opacity": 1,
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#18181b",
                            },
                            {
                              "color": "#71717a",
                              "fontFamily": undefined,
                              "fontSize": 15,
                            },
                            undefined,
                          ]
                        }
                      >
                        email@email.com
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenStack>
  </View>
</RNCSafeAreaProvider>
`;
